### PR TITLE
[MLIR][XeGPU] Add dpas and named barrier ops

### PR DESCRIPTION
**Pull request by chencha3**
This PR adds definition of dpas op, atomic op, named barrier and related ops.

### CODE REVIEW

File: mlir/include/mlir/Dialect/XeGPU/IR/XeGPUOps.td [hunk 1]
```diff
@@ -662,4 +662,152 @@ def XeGPU_UpdateOffsetOp: XeGPU_Op<"update_offset",
...
+    data type, the matrices are `A: vector<8x16xf16>`, `B: vector<16x16xf16>`,
+    and `C/D: vector<8x16xf32>`. Besides the matrix size requirements, DPAS
+    also requires A and B to be loaded with the required data layout. Specially,
+    VNNI layout is required for B operand. It is achieved via setting `vnni_axis = 0`
+    of the corresponding `load_nd` operator. To keep both operands as 3D vector,
+    operand A is loaded via setting `vnni_axis = 1` without impacting the
+    physical layouts change in register. Due to the VNNI transformation, A and B operands
+    are represented as 3D vector, with the last dimension representing the VNNI factor,
+    which is computed as `32/bit_width_of_elem_type`. Therefore, `A: vector<8x16xf16>`
+    is represented as `A: vector<4x8x2xf16>`, and `B:vector<16x16xf16>` is
```
Comment (adam-smnk) [inline mlir/include/mlir/Dialect/XeGPU/IR/XeGPUOps.td:680]:
```suggestion
    is represented as `A: vector<8x8x2xf16>`, and `B: vector<16x16xf16>` is
```

File: mlir/include/mlir/Dialect/XeGPU/IR/XeGPUOps.td [hunk 2]
```diff
@@ -662,4 +662,152 @@ def XeGPU_UpdateOffsetOp: XeGPU_Op<"update_offset",
...
+    represented as `B: vector<8x16x2xf16>`.
+
+    Note: on PVC, the hardware can perform load with VNN transformation when data
```
Comment (adam-smnk) [inline mlir/include/mlir/Dialect/XeGPU/IR/XeGPUOps.td:683]:
```suggestion
    Note: on PVC, the hardware can perform load with VNNI transformation when data
```

File: mlir/include/mlir/Dialect/XeGPU/IR/XeGPUOps.td [hunk 3]
```diff
@@ -662,4 +662,152 @@ def XeGPU_UpdateOffsetOp: XeGPU_Op<"update_offset",
...
+    type($tensorDesc) `,` type($mask) `,` type($value) `->` type($result)
+  }];
+}
+
+def XeGPU_AllocNbarrierOp: XeGPU_Op<"alloc_nbarrier", []> {
+  let summary = "It allocates a set of named barriers.";
+  let description = [{AllocNbarrier is to create a set of named barriers as
+  specified by `nbarrier_num`. Named barriers are workgroup level resources,
+    and are shared by all threads in the workgroup. For example, there are
+    up to 32 barriers (range 0-31) for each Xecore on PVC. A typical use case
```
Comment (adam-smnk) [inline mlir/include/mlir/Dialect/XeGPU/IR/XeGPUOps.td:753]:
```suggestion
    up to 32 barriers (range 0-31) for each XeCore on PVC. A typical use case
```

**Comment by chencha3** [comment]
@adam-smnk @joker-eph Merged #86594, and recreate it againt the main branch. #88973

**Pull request closed without merging**